src/python/pants/core/goals/test.py

```diff
@@ -185,6 +196,12 @@ def register_options(cls, register) -> None:
             help="If a coverage report file is generated, open it on the local system if the "
             "system supports this.",
         )
+        register(
+            "--xml-results",
```

Comment:
I think we'll want to make this name more generic because this option impacts all test runners and not ever test runner uses XML. 

Tansy solved this for coverage by having the generic boolean option `--run-coverage`, then having Pytest Coverage have specific options for the report type and where to write the report in the subsystem `pytest-coverage`: 

https://github.com/pantsbuild/pants/blob/fca3558b3142acdecb5fe5730e6c5d342482e45b/src/python/pants/backend/python/rules/pytest_coverage.py#L218-L238

Here, the boolean option is right, but you'll want to rename the option to something more generic like `save-results` or `write-results`. Also update the `help` message.'

Then, you could add options to https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/python/subsystems/pytest.py to allow specifying where those results should be saved. 

If you want to support multiple formats for those results, you could make that an enum option like Tansy did with report types. If you only care about supporting XML for now, leave off that option for the type of results.

--

Next bit: update `pytest_runner.py` to a) propagate the relevant flags to Pytest, which you already do; then b) capture the relevant folder, which it looks like you also now do.

For splitting up the resulting digest into its relevant pieces, you'll use `SnapshotSubset`:

```python
coverage_digest = await Get[Digest](SnapshotSubset(output_digest, PathGlobs(".coverage"))
results_digest = await Get[Digest](SnapshotSubset(output_digest, PathGlobs(test_results_file))
```

--

Final bit: hook up `test.py` to write the reports to the build root, if any. This needs to happen in the `@goal_rule` because it's a side effect. See below for how we do it with coverage:

https://github.com/pantsbuild/pants/blob/fca3558b3142acdecb5fe5730e6c5d342482e45b/src/python/pants/core/goals/test.py#L291-L321

Your implementation can likely be substantially easier because a) we don't care about grouping related Results files, whereas Coverage needs this so that we can merge multiple individual results into one single report; and b), we don't have an option to open the results.

Your implementation will likely be closer to how we do `binary.py`:

https://github.com/pantsbuild/pants/blob/fca3558b3142acdecb5fe5730e6c5d342482e45b/src/python/pants/core/goals/binary.py#L114-L127

--

Finally, the tests will need some updating. I'm happy to pair on this or to send you some instructions when you're ready.

Reply:
> I think we'll want to make this name more generic because this option impacts all test runners and not ever test runner uses XML.

junit XML is fairly standard at this point, and there are a lot of integrations that can consume it. It might be reasonable to expose it at the top level.